RadixWallet/Features/DappInteractionFeature/Children/AccountsProofOfOwnership/AccountsProofOfOwnership.swift

```diff
@@ -0,0 +1,194 @@
+			dappMetadata: DappMetadata,
+			challenge: DappToWalletInteractionAuthChallengeNonce
+		) {
+			self.accountAddresses = accountAddresses
+			self.dappMetadata = dappMetadata
+			self.challenge = challenge
+		}
+	}
+
+	typealias Action = FeatureAction<Self>
+
+	enum ViewAction: Sendable, Equatable {
+		case task
+		case continueButtonTapped
+	}
+
+	enum InternalAction: Sendable, Equatable {
+		case setAccounts(Accounts)
+		case performSignature(SigningFactors, AuthenticationDataToSignForChallengeResponse)
+	}
+
+	enum DelegateAction: Sendable, Equatable {
+		case provenOwnership([AccountAuthProof], SignedAuthChallenge)
+		case failedToGetAccounts
+		case failedToSign
+	}
+
+	struct Destination: DestinationReducer {
+		@CasePathable
+		enum State: Sendable, Hashable {
+			case signing(Signing.State)
+		}
+
+		@CasePathable
+		enum Action: Sendable, Equatable {
+			case signing(Signing.Action)
+		}
+
+		var body: some ReducerOf<Self> {
+			Scope(state: \.signing, action: \.signing) {
+				Signing()
+			}
+		}
+	}
+
+	@Dependency(\.accountsClient) var accountsClient
+	@Dependency(\.rolaClient) var rolaClient
+	@Dependency(\.factorSourcesClient) var factorSourcesClient
+
+	var body: some ReducerOf<Self> {
+		Reduce(core)
+			.ifLet(destinationPath, action: \.destination) {
+				Destination()
+			}
+	}
+
+	private let destinationPath: WritableKeyPath<State, PresentationState<Destination.State>> = \.$destination
+
+	func reduce(into state: inout State, viewAction: ViewAction) -> Effect<Action> {
+		switch viewAction {
+		case .task:
+			loadAccountsEffect(state: state)
+		case .continueButtonTapped:
+			gatherSignaturePayloadsEffect(state: state)
+		}
+	}
+
+	func reduce(into state: inout State, internalAction: InternalAction) -> Effect<Action> {
+		switch internalAction {
+		case let .setAccounts(accounts):
+			state.accounts = accounts
+			return .none
+		case let .performSignature(signingFactors, authToSignResponse):
+			state.destination = .signing(.init(
+				factorsLeftToSignWith: signingFactors,
+				signingPurposeWithPayload: .signAuth(authToSignResponse)
+			))
+			return .none
+		}
+	}
+
+	func reduce(into state: inout State, presentedAction: Destination.Action) -> Effect<Action> {
+		switch presentedAction {
+		case let .signing(.delegate(signingAction)):
+			switch signingAction {
+			case .cancelSigning:
+				// If the user cancels the signing flow, we just dismiss the `Signing` view and wllow them
+				// to retry by tapping Continue again.
+				state.destination = nil
+				return .none
+
+			case let .finishedSigning(.signAuth(signedAuthChallenge)):
+				state.destination = nil
+
+				var accountsLeftToVerifyDidSign: Set<Account.ID> = Set(state.accounts.map(\.id))
+
+				let accountAuthProofs: [AccountAuthProof] = signedAuthChallenge.entitySignatures.compactMap { signature in
+					guard let account = try? signature.signerEntity.asAccount() else {
+						return nil
+					}
+					accountsLeftToVerifyDidSign.remove(account.id)
```

Comment:
A matter of taste, but you could use create a Set from AccountAuthProofs account ids, and compare it with the `Set(state.accounts.map(\.id)) == Set(accountAuthProofs.map(\.account.id))`

Reply:
I am not only doing the comparison here, but also creating the `WalletToDappInteractionAuthProof` for each signature.